Run feature tests in dynamic Jenkins workers

Sunnatillo opened the pull request:
This PR makes feature tests run directly in dynamic workers instead of using worker as a jumphost

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-centos-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-centos-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-3

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-centos-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-integration-main
/test-centos-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main

Rozzii commented:
/test-ubuntu-integration-main

lentzi90 commented on jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh:
```diff
@@ -0,0 +1,127 @@
+fi
+
+# Fetch k8s logs form target cluster
+target_config=$(sudo find /tmp/ -type f -name "kubeconfig*")
```
Is this file cleaned up between runs? Otherwise we could end up with multiple kubeconfigs!

Sunnatillo replied:
We are using dynamic vm only for one test, then it is deleted.

lentzi90 replied:
Ok good!

lentzi90 commented on jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh:
```diff
@@ -0,0 +1,127 @@
+# Fetch k8s logs form target cluster
+target_config=$(sudo find /tmp/ -type f -name "kubeconfig*")
+if [[ -n "${target_config}" ]]; then
+    #fetch target cluster k8s logs
```
nit
```suggestion
    # fetch target cluster k8s logs
```

lentzi90 reviewed (commented):
/approve

tuminoid commented on jenkins/scripts/dynamic_worker_workflow/run_clean.sh:
```diff
@@ -0,0 +1,18 @@
+else
+    export EPHEMERAL_CLUSTER="minikube"
+fi
+if [[ ("${REPO_NAME}" == "metal3-dev-env") ||
```
One condition per `[[ .. ]]` and no parethesis please.

tuminoid commented on jenkins/scripts/dynamic_worker_workflow/run_clean.sh:
```diff
@@ -0,0 +1,18 @@
+    export EPHEMERAL_CLUSTER="kind"
+else
+    export EPHEMERAL_CLUSTER="minikube"
+fi
```
Add empty line between conditions for readability.

Sunnatillo replied:
added

tuminoid commented on jenkins/scripts/dynamic_worker_workflow/run_clean.sh:
```diff
@@ -0,0 +1,18 @@
+else
+    pushd metal3
+fi
+make clean
```
Add empty line before make for readability.

Sunnatillo replied:
added

tuminoid reviewed (commented):
Small things to fix and some nits.

Rozzii reviewed (changes requested):
/hold
Hold until the nits are fixed.

Sunnatillo commented on jenkins/scripts/dynamic_worker_workflow/fetch_logs.sh:
I copied this most part of the code from normal flow that we have, this would be nice addition, but it would be better this change can be added by separate PR.

tuminoid replied:
Let's do that. This is getting painful to read.

Sunnatillo commented:
@tuminoid @Rozzii I have addressed most of the comments. Regarding implementing a function and other improvements, we need to open new PR  and fix them here and in other integration tests(static worker workflow) as well. 

WDYT?

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main

tuminoid reviewed (commented):
/lgtm

I'm cool with this. Please follow up with the clean up, so we keep it maintainable.

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-ubuntu-integration-main

tuminoid commented:
/unhold
Nits are fixed. @Rozzii please change your review to approving.

Sunnatillo commented:
/test-centos-e2e-feature-main
/test-ubuntu-e2e-feature-main
/test-ubuntu-integration-main

tuminoid reviewed (commented):
/lgtm

Sunnatillo commented:
/test-ubuntu-e2e-feature-main
/test-ubuntu-e2e-feature-release-1-5

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

Sunnatillo commented:
/test-ubuntu-e2e-feature-main

lentzi90 commented:
/test-ubuntu-e2e-feature-main
